src/middleware/auth_middleware.go

```diff
@@ -40,6 +42,9 @@ func TokenAuthMiddleware(tokenProvider auth.TokenProvider) gin.HandlerFunc {
 		}
 		logger := ctxLogger.(*zap.Logger)
 
+		// This is needed by controller-runtime to print logs
+		log.SetLogger(klog.NewKlogr())
+
```

Comment:
I'd rather we use `logger logr.Logger` from `"github.com/go-logr/logr"` to keep it consistent with the rest of the code-base which does something similar:

https://github.com/dana-team/rcs-ocm-deployer/blob/main/test/e2e_tests/helper.go#L27

Reply:
There is no difference in the functionality but I think we should just keep it for consistency sake in the organization code-base